components/board/genericlinux/i2c.go

```diff
@@ -0,0 +1,125 @@
+package genericlinux
+
+import (
+	"context"
+	"encoding/binary"
+	"sync"
+
+	"github.com/edaniels/golog"
+	"periph.io/x/conn/v3/i2c"
+	"periph.io/x/conn/v3/i2c/i2creg"
+	"periph.io/x/host/v3"
+
+	"go.viam.com/rdk/components/board"
+)
+
+func init() {
+	if _, err := host.Init(); err != nil {
+		golog.Global().Debugw("error initializing host", "error", err)
+	}
+}
+
+type i2cBus struct {
+	internal i2c.BusCloser
```

Comment:
TBH I found that name deeply confusing, too, since its main purpose is to act as the bus itself, and not merely a way to close the bus. but our struct is named for being the bus itself, too, hence my dilemma.

I'll rename this `busCloser` and put in a comment explaining that it's more than just for closing.

Reply:
ah I see, perhaps `closeableBus` is the correct choice here?